src/lib/gadgets/foreign-field.unit-test.ts

```diff
@@ -117,11 +199,29 @@ constraintSystem.fromZkProgram(
   )
 );
 
+let mulChain: GateType[] = ['ForeignFieldMul', 'Zero'];
+let mulLayout = ifNotAllConstant(
+  and(
+    contains([mulChain, mrc, mrc, mrc, mrc]),
```

Comment:
which are these mrc's exactly?

I assume the first one is for compact for the remainder,
the second one for quotient range check,
the third one for quotient bound, product1_lo and hi_0,
the last one for result bound, left bound, and right bound

?